### ISSUE TITLE
Rule Change: no-constant-condition should warn about function that is not called

### ISSUE THREAD

**Issue by branko-d** [eslint/eslint#17808]
### What rule do you want to change?

no-constant-condition

### What change do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
function test() {

    const condition = () => {
        return true;
    }

    while (condition) {
        // ...
    }

}
```

### What does the rule currently do for this code?

Passes.

### What will the rule do after it's changed?

Fail on the `while (condition)`, since the function `condition` is always defined and therefore truthy.

The programmer probably meant `while (condition())`.

### Participation

- [ ] I am willing to submit a pull request to implement this change.
